lib/puppet/transaction/event_manager.rb

```diff
@@ -144,20 +144,34 @@ def process_callback(resource, callback, events)
       return false
     end
 
+    message = "Triggered '#{callback}' from #{events.length} events"
+
     resource.send(callback)
 
     if not resource.is_a?(Puppet::Type.type(:whit))
-      resource.notice "Triggered '#{callback}' from #{events.length} events"
+      resource.notice message
+      add_callback_status_event(resource, callback, message, "success")
```

Comment:
nit, if resource is a whit, then we create a message unnecessarily. Suggest moving it into the `if not` block.